notify-send.sh

```diff
@@ -200,109 +200,172 @@ process_special_action() {
     action_key="$1"
     command="$2"
 
-    if [[ -z "$action_key" ]] || [[ -z "$command" ]]; then
+    if [ -z "$action_key" ] || [ -z "$command" ]; then
         echo "Command must not be empty"
         exit 1
     fi
 
     ACTION_COMMANDS=("${ACTION_COMMANDS[@]}" "$action_key" "$command")
 
-    if [[ "$action_key" != close ]]; then
+    if [ "$action_key" != close ]; then
         local action="$(make_action "$action_key" "$name")"
         ACTIONS=("${ACTIONS[@]}" "$action")
     fi
 }
 
 process_posargs() {
-    if [[ "$1" = -* ]] && ! [[ "$positional" = yes ]] ; then
-        echo "Unknown option $1"
-        exit 1
+    case "$1" in
+        -*) if [ "$positional" != yes ] ; then
+                echo "Unknown option $1"
+                exit 1
+            fi ;;
+    esac
+
+    if [ "$SUMMARY_SET" = n ]; then
+        SUMMARY="$1"
+        SUMMARY_SET=y
     else
-        if [[ "$SUMMARY_SET" = n ]]; then
-            SUMMARY="$1"
-            SUMMARY_SET=y
-        else
-            BODY="$1"
-        fi
+        BODY="$1"
     fi
 }
 
-while (( $# > 0 )) ; do
+while [ $# -gt 0 ] ; do
     case "$1" in
         -\?|--help)
             help
             exit 0
             ;;
+
         -v|--version)
             echo "${0##*/} $VERSION"
             exit 0
             ;;
-        -u|--urgency|--urgency=*)
-            [[ "$1" = --urgency=* ]] && urgency="${1#*=}" || { shift; urgency="$1"; }
-            process_urgency "$urgency"
+
+        -u|--urgency)
+            shift
+            process_urgency "$1"
+            ;;
+        --urgency=*)
+            process_urgency "${1#*=}"
             ;;
+
         -t|--expire-time|--expire-time=*)
-            [[ "$1" = --expire-time=* ]] && EXPIRE_TIME="${1#*=}" || { shift; EXPIRE_TIME="$1"; }
-            if ! [[ "$EXPIRE_TIME" =~ ^-?[0-9]+$ ]]; then
-                echo "Invalid expire time: ${EXPIRE_TIME}"
-                exit 1;
-            fi
+            case "$1" in
+                --expire-time=*) EXPIRE_TIME="${1#*=}" ;;
+                *) shift; EXPIRE_TIME="$1" ;;
+            esac
```

Comment:
This is a prime example of what I just mentioned, where we need the extra case to check the different variant so we don't inadvertently shift arguments or repeat option parsing/validation logic in a different branch just for the `--option=*` variant.